src/ExtEventLoop.php

```diff
@@ -25,13 +25,31 @@ class ExtEventLoop implements LoopInterface
     private $readListeners = [];
     private $writeListeners = [];
     private $running;
+    private $signals;
+    private $signalEvents = [];
 
     public function __construct(EventBaseConfig $config = null)
     {
         $this->eventBase = new EventBase($config);
         $this->futureTickQueue = new FutureTickQueue();
         $this->timerEvents = new SplObjectStorage();
 
+        $this->signals = new SignalsHandler();
+        $this->signals->on('on', function ($signal) {
+            $this->signalEvents[$signal] = Event::signal($this->eventBase, $signal, function () use ($signal) {
+                $this->signals->call($signal);
+            });
+            $this->signalEvents[$signal]->add();
+        });
+        $this->signals->on('off', function ($signal) {
+            $this->futureTick(function () use ($signal) {
```

Comment:
It does, the issue is that certain loops throw errors when you try to remove a signal handler while handling the given signal. In order to mitigate the race condition each loop does a check to ensure there are no registered signal handlers for the given signal. Should add documentation for it.